workspaces/vscode/pl-gosu-helper.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# The purpose of this script is to prevent subtle issues that might occur if
+# you are testing in the local PrairieLearn Docker container and it attempts
+# to launch workspaces as root.
```

Comment:
This is actually a launch script we've been using in some of the other images (see `workspaces/rstudio` and `workspaces/desktop`), courtesy of @echuber2.  Currently, workspace containers launch with root on local dev instead of the user they are supposed to be, which can lead to issues with permissions and things working correctly on local and not on prod, or vice-versa.  
I suppose it would be more appropriate to fix this in PL, rather than having these workarounds in all of the containers (but trying to work through the workspace logic is kind of scary 😢 )

Reply:
> I suppose it would be more appropriate to fix this in PL, rather than having these workarounds in all of the containers (but trying to work through the workspace logic is kind of scary 😢 )

Yeah, this absolutely sounds like the right thing to do. By "the workspace logic", are you referring to everything in `workspace_host`? Agree that that's kinda hairy, but if you have a good sense of the abstract things that need to be done, it'd be awesome if you could open an issue capturing that.

Reply:
This was a pretty ad-hoc solution that worked for Dirk's rstudio situation and Elsa's OCaml (VS Code) situation. I agree it would be good if the local PL docker always tried to run workspaces as 1001 to emulate what the server does, but maybe some staff users' local OS won't allow that for whatever reason.

I think it would work to have PL try to run workspaces as 1001 by default, and if the OS won't allow it, then to issue a verbose error message to the local session terminal along with offering an override flag.